pkg/util/liveconfig/hive.go

```diff
@@ -21,14 +23,46 @@ const (
 	hiveAdoptEnableEnvVar     = "ARO_ADOPT_BY_HIVE"
 )
 
-func parseKubeconfig(credentials []mgmtcontainerservice.CredentialResult) (*rest.Config, error) {
-	res := make([]byte, base64.StdEncoding.DecodedLen(len(*credentials[0].Value)))
-	_, err := base64.StdEncoding.Decode(res, *credentials[0].Value)
+func getAksKubeconfig(ctx context.Context, managedClustersClient containerservice.ManagedClustersClient, location string, shard int) (*rest.Config, error) {
+	aksClusterName := fmt.Sprintf("aro-aks-cluster-%03d", shard)
+
+	aksClusters, err := managedClustersClient.List(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	clientconfig, err := clientcmd.NewClientConfigFromBytes(res)
+	var aksCluster *mgmtcontainerservice.ManagedCluster
+
+outerLoop:
+	for aksClusters.NotDone() {
+		for _, cluster := range aksClusters.Values() {
+			if *cluster.Name == aksClusterName && *cluster.Location == location {
+				aksCluster = &cluster
+				break outerLoop
+			}
+		}
+		err = aksClusters.NextWithContext(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
it would be nicer to have that in a function so we don't have to use loop annotations

Reply:
Good call.